File: Homework/HW1/HW1.java
public class HW1 {
    public static boolean isPalindrome(String s) {
        String lowerCaseString = s.toLowerCase();

        StringBuilder cleanString = new StringBuilder();
        for (int i = 0; i < lowerCaseString.length(); i++) {
            char c = lowerCaseString.charAt(i);
            if (Character.isLetterOrDigit(c)) {
                cleanString.append(c);
            }
        }

        int left = 0;
        int right = cleanString.length() - 1;
        while (left < right) {
            if (cleanString.charAt(left) != cleanString.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    public static void main(String[] args) {
        String testString1 = "No lemon, no melon";
        System.out.println(isPalindrome(testString1));

        String testString2 = "12321";
        System.out.println(isPalindrome(testString2));

    }
}

File: Homework/HW1/HW1.py
class HW1:
    def is_palindrome(s):
        lower_case_string = s.lower()
        clean_string = ''.join(c for c in lower_case_string if c.isalnum())

        left, right = 0, len(clean_string) - 1
        while left < right:
            if clean_string[left] != clean_string[right]:
                return False
            left += 1
            right -= 1
        return True


if __name__ == "__main__":
    test_string1 = "No lemon, no melon"
    print(HW1.is_palindrome(test_string1))

    test_string2 = "12321"
    print(HW1.is_palindrome(test_string2))
